### DISCUSSION TITLE
Password access for QuartoPub

### DISCUSSION THREAD

**Discussion by cjsrussell** [quarto-dev/quarto-cli#3669] [Q&A] [answered]
I think it would be great to have a simple password lock for Quarto Pub documents.

I know the principle is that QuartoPub is a place to host public docs, but this would be a useful feature for soft launching documents to say other members of one's team (esp less tech savvy folks) before making them publicly available.

Just a thought! Thanks for all your work on this, its a useful resource.

**Comment by mcanouil** [comment, marked as answer]
Just wondering, but it might be possible to use in your `_quarto.yml` or Quarto header:

```
resources:
  - .htaccess
  - .htpasswd
```

See for example most or any of resources you can find with <https://duckduckgo.com/?q=htaccess+password>

**Reply by udurraniAtPresage** [reply]
I tried your suggestion for password protection of a quarto pub document with a quarto dashboard. But the published dashboard is not password protected.

- Dashboard url: https://presage.quarto.pub/test/
- GitHub repo: https://github.com/udurraniAtPresage/example-dashboard

Am I missing anything in the `.htaccess` and `.htpasswd` files? @cjsrussell were you successful in implementing password protection? Thanks.

**Reply by mcanouil** [reply]
I'm afraid that if you want to host private document, you'll have to find another cloud provider.
Quarto Pub does not support the use of `.htpasswd`.

**Reply by udurraniAtPresage** [reply]
Thanks for the clarification. I even tried github pages but got the same result.

**Reply by mcanouil** [reply]
GitHub Pages only serve static files, you have no access nor cannot change server configuration which `.ht*` files do. (Some options in `.htaccess` might still work)
All cloud providers like this won't allow you to use those files.
